packages/cli-kit/src/public/node/ui.tsx

```diff
@@ -48,10 +48,15 @@ export async function renderConcurrent({renderOptions, ...props}: RenderConcurre
     ...props,
   }
   if (terminalSupportsRawMode(renderOptions?.stdin)) {
-    return render(<ConcurrentOutput {...newProps} />, {
-      ...renderOptions,
-      exitOnCtrlC: typeof props.onInput === 'undefined',
-    })
+    try {
+      await render(<ConcurrentOutput {...newProps} />, {
+        ...renderOptions,
+        exitOnCtrlC: typeof props.onInput === 'undefined',
+      })
+    } catch (error) {
+      newProps.abortController.abort()
```

Comment:
On a second pass, the only reason I added this is because the parent can also be `renderConcurrent` itself if abortController is not passed form the outside. I think we should make `abortController` optional in renderConcurrent and remove this.